temporalio/client.py

```diff
@@ -1112,12 +1112,12 @@ async def get_worker_task_reachability(
 class ClientConfig(TypedDict, total=False):
     """TypedDict of config originally passed to :py:meth:`Client`."""
 
-    service_client: temporalio.service.ServiceClient
-    namespace: str
-    data_converter: temporalio.converter.DataConverter
-    interceptors: Sequence[Interceptor]
-    default_workflow_query_reject_condition: Optional[
-        temporalio.common.QueryRejectCondition
+    service_client: Required[temporalio.service.ServiceClient]
+    namespace: Required[str]
```

Comment:
huh, weird, you think Required would be the default...

Reply:
Yes, there's a fair amount of reading material on the topic for anyone so-inclined.
https://peps.python.org/pep-0655/
https://typing.readthedocs.io/en/latest/spec/typeddict.html#required-notrequired

Reply:
Hrmm, I am trying to remember why I did `total=False` here. I wonder if I expected someone to do something like:

```python
extra: client.ClientConfig = { "interceptors": [MyInterceptor()] }

# ...

new_client = Client(my_service_client, **extra)
```

This would break that use case. If we are ok breaking that strange use case (I am), then I wonder if we should just remove `total=False`. I can't think of a reason that `total=False` should be there if every field is `Required`, can anyone else?

Reply:
Hm, I'm not sure I should have merged without resolving this conversation. May need to revisit this. E.g. in features repo we attempt to do 
```python
register_feature(
    workflows=[Workflow],
    check_result=check_result,
    start=start,
    additional_client_config=ClientConfig(interceptors=[MyClientInterceptor()]),
)
```


https://github.com/temporalio/features/blob/bump-sdk-version/bin/main/update/client_interceptor/feature.py#L66

which [isn't passing type checks](https://github.com/temporalio/features/actions/runs/11282286622/job/31379414486?pr=549).